lib/iris/fileformats/nimrod_load_rules.py

```diff
@@ -25,58 +28,306 @@
+#  3: ['sigma', 'model level'],
+#  4: ['eta', 'model level'],
+#  5: ['radar beam number', 'unknown'],
+#  7: ['potential temperature', 'unknown'],
+#  8: ['equivalent potential temperature', 'unknown'],
+#  9: ['wet bulb potential temperature', 'unknown'],
+#  10: ['potential vorticity', 'unknown'],
+#  11: ['cloud boundary', 'unknown'],
+
+SOIL_TYPE_CODES = {
+    1: "broadleaf_tree",
+    2: "needleleaf_tree",
+    3: "c3_grass",
+    4: "c4_grass",
+    5: "crop",
+    6: "shrub",
+    7: "urban",
+    8: "water",
+    9: "soil",
+    10: "ice",
+    601: "urban_canyon",
+    602: "urban_roof",
+}
+TIME_AVERAGING_CODES = {
+    8192: "maximum in period",
+    4096: "minimum in period",
+    2048: "unknown(2048)",
+    1024: "unknown(1024)",
+    512: "time lagged",
+    256: "extrapolation",
+    128: "accumulation or average",
+    64: "from UM 150m",
+    32: "scaled to UM resolution",
+    16: "averaged over multiple surface types",
+    8: "only observations used",
+    4: "smoothed",
+    2: "cold bias applied",
+    1: "warm bias applied",
+}
 
 
 class TranslationWarning(Warning):
     pass
 
 
-def name(cube, field):
-    """Set the cube's name from the field."""
+def is_missing(field, value):
+    """Returns True if value matches an "is-missing" number."""
+    return any(
+        np.isclose(value, [field.int_mdi, field.float32_mdi, NIMROD_DEFAULT])
+    )
 
-    cube.rename(field.title.strip())
+
+def name(cube, field):
+    """Set the cube's name from the field.
+    Modifies the Nimrod object title based on other meta-data in the
+    Nimrod field and known use cases.
+    """
+    if field.field_code == 12:
+        field.title = "air_pressure"
+    if field.field_code == 27:
+        field.title = "snow fraction"
+    if field.field_code == 28:
+        field.title = "snow probability"
+    if field.field_code == 29 and field.threshold_value >= 0.0:
+        field.title = "fog fraction"
+    if field.field_code == 58:
+        field.title = "temperature"
+    if field.field_code == 61:
+        field.title = "precipitation"
+    if field.field_code == 63:
+        field.title = "precipitation"
+    if field.field_code == 101:
+        field.title = "snow_melting_level_above_sea_level"
+    if field.field_code == 102:
+        field.title = "rain_melted_level_above_sea_level"
+    if field.field_code == 155:
+        field.title = "Visibility"
+    if field.field_code == 156:
+        field.title = "Worst visibility in grid point"
+    if field.field_code == 161 and field.threshold_value >= 0.0:
+        field.title = "minimum_cloud_base_above_threshold"
+    if field.field_code == 218:
+        field.title = "snowfall"
+    if field.field_code == 172:
+        field.title = "cloud_area_fraction_in_atmosphere"
+    if field.field_code == 421:
+        field.title = "precipitation type"
+    if field.field_code == 501:
+        field.title = "vector_wind_shear"
+        field.source = ""
```

Comment:
This one looks a bit odd, as a special case.
Definitely needed ?

Reply:
Yes, I found a mistake in the code that generates these and it's wrong in the archive. Much easier to handle here.

Reply:
Ok!